### PR TITLE
Add language tag to code blocks and correct indented code block

### PR DESCRIPTION
**Pull request by bact**
- Add language tag (like `xml`, `json`) to Markdown code blocks, so the code syntax will be properly highlighted in the document
- Fix indent of a code block nested inside a numbered list
- Fix other Markdown lint warnings. Like removing end-of-line trailing spaces and ensuring a blank line before and after a heading.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,13 +30,13 @@ Please check the [CLI Reference](docs/sbom-tool-cli-reference.md) document for a
 
 ##### WinGet
 
-```sh
+```powershell
```
Comment (DaveTryon) [inline README.md:33]: `winget` is windows-specific, but not powershell specific.
```suggestion
```shell
```

File: README.md [hunk 2]
```diff
@@ -30,13 +30,13 @@ Please check the [CLI Reference](docs/sbom-tool-cli-reference.md) document for a
...
 winget install Microsoft.SbomTool
 ```
 
 ##### Homebrew
 
-```bash
+```shell
```
Comment (DaveTryon) [inline README.md:39]: Visually identical, and bash is appropriate for any homebrew environment (see https://github.com/highlightjs/highlight.js/blob/main/SUPPORTED_LANGUAGES.md)

File: README.md [hunk 3]
```diff
@@ -50,14 +50,14 @@ Invoke-WebRequest -Uri "https://github.com/microsoft/sbom-tool/releases/latest/d
 
 ##### Linux (curl)
 
-```bash
+```shell
```
Comment (DaveTryon) [inline README.md:53]: Visually identical, and bash is appropriate for any linux environment (see https://github.com/highlightjs/highlight.js/blob/main/SUPPORTED_LANGUAGES.md)

File: README.md [hunk 4]
```diff
@@ -50,14 +50,14 @@ Invoke-WebRequest -Uri "https://github.com/microsoft/sbom-tool/releases/latest/d
...
 curl -Lo sbom-tool https://github.com/microsoft/sbom-tool/releases/latest/download/sbom-tool-linux-x64
 chmod +x sbom-tool
 ```
 
 ##### macOS (curl)
 
-```bash
+```shell
```
Comment (DaveTryon) [inline README.md:60]: Visually identical, and bash is appropriate for any macos environment (see https://github.com/highlightjs/highlight.js/blob/main/SUPPORTED_LANGUAGES.md)

File: README.md [hunk 5]
```diff
@@ -94,7 +94,7 @@ Please check the [API Reference](docs/sbom-tool-api-reference.md) document for a
 
 Once you have installed the command line tool for your OS, run the tool using this command:
 
-```
+```shell
```
Comment (DaveTryon) [inline README.md:97]: In this case, we want the < and > to be placeholders, not literals, so we want the generic formatting

File: README.md [hunk 6]
```diff
@@ -108,7 +108,7 @@ A more detailed list of available CLI arguments for the tool can be found [here]
 
 With an SBOM file in hand, use the tool to validate the output file with the command:
 
-```
+```shell
```
Comment (DaveTryon) [inline README.md:111]: In this case, we want the < and > to be placeholders, not literals, so we want the generic formatting

File: README.md [hunk 7]
```diff
@@ -124,11 +124,11 @@ Currently only SPDX2.2 is supported.
 
 Use the tool to redact any references to files from a given SBOM or set of SBOMs with either of the following commands:
 
-```
+```shell
```
Comment (DaveTryon) [inline README.md:127]: In this case, we want the < and > to be placeholders, not literals, so we want the generic formatting

File: README.md [hunk 8]
```diff
@@ -124,11 +124,11 @@ Currently only SPDX2.2 is supported.
...
 sbom-tool redact -sd <directory containing SBOMs to redact> -o <output path>
 ```
 
-```
+```shell
```
Comment (DaveTryon) [inline README.md:131]: In this case, we want the < and > to be placeholders, not literals, so we want the generic formatting

File: docs/building-from-source.md [hunk 9]
```diff
@@ -37,13 +37,13 @@ A good new-user tutorial for Visual Studio is available at [this](https://www.yo
...
 
-```
+```shell
 dotnet build
 ```
 
 Users can run the sbom tool using this command which contains the minimum required set of parameters:
 
-```
+```shell
```
Comment (DaveTryon) [inline docs/building-from-source.md:46]: In this case, we want the < and > to be placeholders, not literals, so we want the generic formatting

File: docs/sbom-tool-cli-reference.md [hunk 10]
```diff
@@ -22,7 +22,7 @@ dotnet build --output c:\outputDrop
 
 Now the user can generate a SBOM for the above project by running the tool they just downloaded:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:25]: This isn't powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 11]
```diff
@@ -51,7 +51,7 @@ The above list contains the minimum mandatory parameters that the user needs to
 
 By default, the tool will generate SBOM file in a newly created subfolder called `_manifest` inside the BuildDropPath (-b).  In case the user wants to place the SBOM in a different path, specify the `ManifestDirPath -m` parameter, e.g.:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:54]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 12]
```diff
@@ -63,35 +63,35 @@ This command will cause the SBOM tool to generate the SBOM inside the `c:\sboms`
 
 The user can specify verbose logging just by specifying the -V parameter, e.g:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:66]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 13]
```diff
@@ -63,35 +63,35 @@ This command will cause the SBOM tool to generate the SBOM inside the `c:\sboms`
...
 ./sbom-tool-win-x64.exe generate -b c:\outputDrop -bc c:\Users\test\TestProject -pn TestProject -pv 1.0.0 -ps MyCompany -nsb http://mycompany.com -V Verbose
 ```
 
 ### Scan docker images for dependency packages
 
 Users can scan docker images in order to determine dependency packages.  In this scenario, the user wants to gather dependencies from the docker image `testImage:0.0.1`.  The user can run the following command:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:74]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 14]
```diff
@@ -63,35 +63,35 @@ This command will cause the SBOM tool to generate the SBOM inside the `c:\sboms`
...
 ./sbom-tool-win-x64.exe generate -b c:\outputDrop -bc c:\Users\test\TestProject -pn TestProject -pv 1.0.0 -ps MyCompany -nsb http://mycompany.com -di testImage:0.0.1
 ```
 
 In addition to the test image, the user may also want to gather all dependencies in a build machine named `ubuntu:1.9`.  The command can specify multiple image arguments for this parameter by separating them with a comma:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:80]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 15]
```diff
@@ -63,35 +63,35 @@ This command will cause the SBOM tool to generate the SBOM inside the `c:\sboms`
...
 ./sbom-tool-win-x64.exe generate -b c:\outputDrop -bc c:\Users\test\TestProject -pn TestProject -pv 1.0.0 -ps MyCompany -nsb http://mycompany.com -di testImage:0.0.1,ubuntu:1.9
 ```
 
 The arguments for `-b` and `-bc` will specify the path that the tool will scan. For example, the user can generate an SBOM for only the dependency packages of the Docker image with the command:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:86]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 16]
```diff
@@ -63,35 +63,35 @@ This command will cause the SBOM tool to generate the SBOM inside the `c:\sboms`
...
 ./sbom-tool-win-x64.exe generate -m c:\outputPath -pn TestProject -pv 1.0.0 -ps MyCompany -nsb http://mycompany.com -di testImage:0.0.1
 ```
 
 `-m` provides the user-defined path for generating the SBOM. The tool will generate a new `_manifest\spdx_2.2` subfolder which will hold the SBOM file created during this run. The files section for these parameters will be empty as this run will only scan for the dependency packages of the image.
 
 In order to scan a path to populate the files section of the SBOM, the user can run the following command:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:94]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 17]
```diff
@@ -100,29 +100,29 @@ In order to scan a path to populate the files section of the SBOM, the user can
 You can exclude directories from the component scan by specifying the `-cd` parameter you can pass arguments directly to Component Detection. One of these arguments is `--DirectoryExclusionList`  Filters out specific directories following a minimatch pattern from the component scan which will leave
 the contents of these directories out of the packages section of the SBOM. For example, if you wanted to exclude the `bin` directory from the component scan you would run the following command
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:103]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 18]
```diff
@@ -100,29 +100,29 @@ In order to scan a path to populate the files section of the SBOM, the user can
...
 ./sbom-tool-win-x64.exe generate -b c:\outputDrop -bc c:\Users\test\TestProject -pn TestProject -pv 1.0.0 -ps MyCompany -nsb http://mycompany.com -cd "--DirectoryExclusionList **/bin/**"
 ```
 
 You can give multiple exclusion patterns by repeating the `--DirectoryExclusionList` argument. (Note that minimatch combines like `**/bin/**|**/obj/**` won't work):
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:109]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

File: docs/sbom-tool-cli-reference.md [hunk 19]
```diff
@@ -100,29 +100,29 @@ In order to scan a path to populate the files section of the SBOM, the user can
...
 ./sbom-tool-win-x64.exe generate -b c:\outputDrop -bc c:\Users\test\TestProject -pn TestProject -pv 1.0.0 -ps MyCompany -nsb http://mycompany.com -t c:\telemetry
 ```
 
 ## Validating an SBOM
 
 With a SBOM file in hand, use the tool to validate the output file with the command:
 
-```
+```powershell
```
Comment (DaveTryon) [inline docs/sbom-tool-cli-reference.md:125]: This is windows-specific but not powershell-specific
```suggestion
```shell
```

**Comment by bact** [comment]
Thank you for detailed suggestions. I have updated them accordingly.

**Review by DaveTryon — APPROVED** [review]

**Pull request merged**